src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2513,6 +2515,360 @@ public Expand(Func<TIn, IEnumerator<TOut>> extrapolate)
         public override string ToString() => "Expand";
     }
 
+    public sealed class NotYetThereSentinel : Exception
+    {
+        public static readonly NotYetThereSentinel Instance = new();
+    }
+
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <typeparam name="TIn">TBD</typeparam>
+    /// <typeparam name="TOut">TBD</typeparam>
+    [InternalApi]
+    public sealed class
+        SelectValueTaskAsync<TIn, TOut> : GraphStage<FlowShape<TIn, TOut>>
+    {
+        #region internal classes
+
+        private sealed class Logic : InAndOutGraphStageLogic
+        {
+            private sealed class Holder<T>
+            {
+                public Result<T> Element { get; private set; }
+                private readonly Action<Holder<T>> _callback;
+                private ValueTask<T> _pending;
```

Comment:
It seemed to me, that the best thing to do here was have `Holder<T>` itself keep the continuation and just re-use them appropriately.

Yes, it costs the size of `ValueTask<T>` but we'd take at least that `.AsTask()`ing and in this case we only pay it 'once'.

Reply:
Seems pretty clever to me - I'd love to see a benchmark comparison between this and a normal `SelectAsync`